File: VECTOAux/VectoAuxiliaries/Hvac/ITechListBenefitLine.vb
Namespace Hvac
	Public Interface ITechListBenefitLine
		Property Units As string
		Property Category As String
		Property BenefitName As String

		Property LowFloorH As double
		Property LowFloorV As double
		Property LowFloorC As double

		Property SemiLowFloorH As double
		Property SemiLowFloorV As double
		Property SemiLowFloorC As double

		Property RaisedFloorH As double
		Property RaisedFloorV As double
		Property RaisedFloorC As double

		Property OnVehicle As Boolean
		Property ActiveVH As Boolean
		Property ActiveVV As Boolean
		Property ActiveVC As Boolean
		Property LineType As TechLineType

		ReadOnly Property H As Double
		ReadOnly Property VH As Double
		ReadOnly Property VV As Double
		ReadOnly Property VC As Double
		ReadOnly Property C As Double

		Sub CloneFrom(source As ITechListBenefitLine)

		Function IsEqualTo(source As ITechListBenefitLine) As Boolean
	End Interface
End Namespace

File: VECTOAux/BusAuxiliaries/Interfaces/DownstreamModules/HVAC/ITechListBenefitLine.cs
namespace DownstreamModules.HVAC {
	public interface ITechListBenefitLine
	{
		string Units { get; set; }
		string Category { get; set; }
		string BenefitName { get; set; }

		double LowFloorH { get; set; }
		double LowFloorV { get; set; }
		double LowFloorC { get; set; }

		double SemiLowFloorH { get; set; }
		double SemiLowFloorV { get; set; }
		double SemiLowFloorC { get; set; }

		double RaisedFloorH { get; set; }
		double RaisedFloorV { get; set; }
		double RaisedFloorC { get; set; }

		bool OnVehicle { get; set; }
		bool ActiveVH { get; set; }
		bool ActiveVV { get; set; }
		bool ActiveVC { get; set; }
		TechLineType LineType { get; set; }

		double H { get; }
		double VH { get; }
		double VV { get; }
		double VC { get; }
		double C { get; }

		void CloneFrom(ITechListBenefitLine source);

		bool IsEqualTo(ITechListBenefitLine source);
	}
}